src/library/scala/collection/Iterator.scala

```diff
@@ -433,6 +434,8 @@ trait Iterator[+A] extends IterableOnce[A] with IterableOnceOps[A, Iterator, Ite
 
   final def length: Int = size
 
+  @inline final override def isEmpty: Boolean = !hasNext
```

Comment:
This should possibly be deprecatedOverriding instead in case someone did it the other way around?  (`override def isEmpty = myLogic; def hasNext = !isEmpty`).

Reply:
Not sure how removing 'final' affects optimization. Generally it shouldn't be a big deal for people to switch their impl to override hasNext in the process of upgrading to 2.13 but _shrug_.

Reply:
It would prevent inlining but allow for a smoother migration. We could deprecate it now and the make it `inline final` in 2.14.

Reply:
ok, done.